core/contracts/oracle/implementation/Voting.sol

```diff
@@ -363,8 +363,14 @@ contract Voting is Testable, Ownable, OracleInterface, VotingInterface, Encrypte
         // 0 hashes are disallowed in the commit phase, so they indicate a different error.
         // Cannot reveal an uncommitted or previously revealed hash
         require(voteSubmission.commit != bytes32(0), "Invalid hash reveal");
-        // Committed hash doesn't match revealed price and salt
-        require(keccak256(abi.encode(price, salt)) == voteSubmission.commit, "Invalid commit hash & salt");
+        require(keccak256(abi.encodePacked(
+            price,
+            salt,
+            msg.sender,
+            time,
+            roundId,
+            identifier
+        )) == voteSubmission.commit, "Commit hash must match px, salt, address, timestamp, round, and identifier");
```

Comment:
Require messages longer than 32 bytes take up way more storage. We should try keep them below 32 for consistency. Can I propose the following require message which is below the 32 char limit?
```suggestion 
        )) == voteSubmission.commit, "Revealed data != commit hash");
```